app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  has_many :likes
+  has_many :comments
+
+  belongs_to :author, class_name: 'User'
+
+  def post_counter
+    author.increment!(:posts_counter)
+  end
+
```

Comment:
[OPTIONAL]

- It would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#post_counter` method every time a new post has been saved to the database, by this you can make the `post_counter` method private which will increase the security.

```ruby
# app/models/post.rb
  after_save :post_counter

  private

  def post_counter
    author.increment!(:posts_counter)
  end
```